packages/amplify-provider-awscloudformation/src/aws-utils/aws-cognito-client.ts

```diff
@@ -0,0 +1,22 @@
+import { $TSContext } from 'amplify-cli-core';
+import aws from './aws.js';
+import * as AWS from 'aws-sdk';
+import { AwsSecrets, loadConfiguration } from '../configuration-manager';
+
+export class CognitoUserPoolClientProvider {
+  private static instance: CognitoUserPoolClientProvider;
+  readonly client: AWS.CognitoIdentityServiceProvider;
+
+  static async getInstance(context: $TSContext, options = {}): Promise<CognitoUserPoolClientProvider> {
+    if (!CognitoUserPoolClientProvider.instance) {
```

Comment:
nit: do we need any kind of mutex synchronization mechanisms here? I didn't see it in any other singleton we've implemented, but this isn't async safe.

Reply:
Can you explain me how this isnt async safe. If this method is static , so it can be only called by class and it can be awaited